src/fenicsxconcrete/finite_element_problem/base_material.py

```diff
@@ -49,34 +51,35 @@ def __init__(self,
 
         # set log level...
-        if self.p['log_level'] == 'DEBUG':
-            df.log.LogLevel(10)
-            logging.getLogger("FFC").setLevel(logging.DEBUG)
-            logging.getLogger("UFL").setLevel(logging.DEBUG)
-            logger.add(sys.stderr, level="DEBUG")
-        elif self.p['log_level'] == 'INFO':
-            df.log.LogLevel(20)
-            logging.getLogger("FFC").setLevel(logging.INFO)
-            logging.getLogger("UFL").setLevel(logging.INFO)
-            logger.add(sys.stderr, level="INFO")
-        elif self.p['log_level'] == 'WARNING':
-            df.log.LogLevel(30)
-            logging.getLogger("FFC").setLevel(logging.WARNING)
-            logging.getLogger("UFL").setLevel(logging.WARNING)
-            logger.add(sys.stderr, level="WARNING")
-        elif self.p['log_level'] == 'ERROR':
-            df.log.LogLevel(40)
-            logging.getLogger("FFC").setLevel(logging.ERROR)
-            logging.getLogger("UFL").setLevel(logging.ERROR)
-            logger.add(sys.stderr, level="ERROR")
-        elif self.p['log_level'] == 'CRITICAL':
-            df.log.LogLevel(50)
-            logging.getLogger("FFC").setLevel(logging.CRITICAL)
-            logging.getLogger("UFL").setLevel(logging.CRITICAL)
-            logger.add(sys.stderr, level="CRITICAL")
-        else:
-            level = self.p['log_level']
-            raise ValueError(f'unknown log level {level}')
+        self.logger.setLevel(self.p['log_level'])
+        # if self.p['log_level'] == 'DEBUG':
+        #     df.log.LogLevel(10)
+        #     logging.getLogger("FFC").setLevel(logging.DEBUG)
+        #     logging.getLogger("UFL").setLevel(logging.DEBUG)
+        #     logger.add(sys.stderr, level="DEBUG")
+        # elif self.p['log_level'] == 'INFO':
+        #     df.log.LogLevel(20)
+        #     logging.getLogger("FFC").setLevel(logging.INFO)
+        #     logging.getLogger("UFL").setLevel(logging.INFO)
+        #     logger.add(sys.stderr, level="INFO")
+        # elif self.p['log_level'] == 'WARNING':
+        #     df.log.LogLevel(30)
+        #     logging.getLogger("FFC").setLevel(logging.WARNING)
+        #     logging.getLogger("UFL").setLevel(logging.WARNING)
+        #     logger.add(sys.stderr, level="WARNING")
+        # elif self.p['log_level'] == 'ERROR':
+        #     df.log.LogLevel(40)
+        #     logging.getLogger("FFC").setLevel(logging.ERROR)
+        #     logging.getLogger("UFL").setLevel(logging.ERROR)
+        #     logger.add(sys.stderr, level="ERROR")
+        # elif self.p['log_level'] == 'CRITICAL':
+        #     df.log.LogLevel(50)
+        #     logging.getLogger("FFC").setLevel(logging.CRITICAL)
+        #     logging.getLogger("UFL").setLevel(logging.CRITICAL)
+        #     logger.add(sys.stderr, level="CRITICAL")
+        # else:
+        #     level = self.p['log_level']
+        #     raise ValueError(f'unknown log level {level}')
```

Comment:
```suggestion
```
I would suggest to remove this. 
Not sure was `df.log.LogLevel(50)` does exactly, but I don't think it has the desired effect.
```
 ❯ python
Python 3.10.9 | packaged by conda-forge | (main, Feb  2 2023, 20:20:04) [GCC 11.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import dolfinx.log as log
>>> log.LogLevel(50)
<LogLevel.???: 50>
>>> log.get_log_level()
<LogLevel.WARNING: -1>
```

I would suggest to not set anything here that is related to the components of fenicsx. IMO the user should have full control over each logger and what level is needed for the application.
See `tests/test_basic_logging.py` for how this could be done.